### PR TITLE
MAD-14978-apply-psr-12-coding-standard-core

### PR DESCRIPTION
**Pull request by mdk999**
- Updated README.md
- Changed to PSR12
- Added PSR12 Sniffs as warnings for visibility

### CODE REVIEW

**Review by davemednick — APPROVED** [review]

**Review by mclorch — COMMENTED** [review]
do we need a phpcs repo if we are gonna default to the standard? i.e. could this repo go away instead?

File: .php-cs-fixer.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...jects\/nocworx-lib\/src\/NocWorx\/Lib\/Console\/Creator\/CreateCommand.php":"78ee1645e1fa22f6e64f2e39fc2a8ddb","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Console\/Creator\/Creator.php":"710b725b2079375c58a9b2c379b9c86c","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Console\/Application.php":"cf7550daf19240af407dab9ad5b581e9","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Console\/CommandDiscoveryFactory.php":"79058b55d35040e254291cc5f16c4e85","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Console\/Command.php":"8446c5a9fe2b9043afedc3bef7ace0d0","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/ComposerHooks.php":"7346a9a05203905d777614c867a65ef7","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Guzzle\/RetryMiddleware.php":"f4a27e9553b24db90bbc41c174904e7a","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Process.php":"bd580506deb53ac44c57255d820f43f1","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Session\/Manager.php":"5e563aa07a57fd594417b584c9a9f77a","\/Users\/weaponx\/Projects\/nocworx-lib\/src\/NocWorx\/Lib\/Session\/Exception.php":"e1fc34fa6bc2bc716c3cc5e625a48283"}}
```
Comment (mclorch) [inline .php-cs-fixer.cache:1]: add to `.gitignore`

File: .php-cs-fixer.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    'backtick_to_shell_exec' => true,
+    'braces' => true,
+    'indentation_type' => true,
+    'method_chaining_indentation' => true,
+    'Symplify/codingstandard' => true,
+  ]
+)
+  ->setFinder($finder)->setIndent('  ');
+
+/*$finder = (new PhpCsFixer\Finder())
```
Comment (mclorch) [inline .php-cs-fixer.php:45]: remove commented out code

**Review by mclorch — APPROVED** [review]

File: .php-cs-fixer.php (hunk 2 above)
Comment (mdk999) [reply .php-cs-fixer.php]: Done

File: .php-cs-fixer.cache (hunk 1 above)
Comment (mdk999) [reply .php-cs-fixer.cache]: done

File: composer.lock [hunk 3]
```diff
@@ -151,24 +151,24 @@
         },
         {
             "name": "composer/semver",
-            "version": "3.4.2",
```
Comment (mclorch) [inline composer.lock]: should there be accompanying composer.json changes here? e.g. how does composer.lock get updated without composer.json changes?

Comment (mdk999) [reply composer.lock]: I did a composer update but there were no composer.json changes only changes in the lock file

Comment (mclorch) [reply composer.lock]: I see - that probably makes sense!

File: composer.json [hunk 4]
```diff
@@ -4,9 +4,27 @@
...
+    "symplify/coding-standard": "dev-main",
+    "slevomat/coding-standard": "^8.15",
+    "friendsofphp/php-cs-fixer": "^3.64",
+    "symfony/dependency-injection": "^7.1",
+    "symfony/config": "^7.1"
   },
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/mdk999/coding-standard"
```
Comment (mclorch) [inline composer.json:17]: should this be mdk999 or something else?

Comment (mclorch) [reply composer.json:17]: this fork should be pulled into nocworx organization instead - @mpedrummer will take care of that

Comment (davemednick) [reply composer.json:17]: Yeah - what's up with this? Shouldn't you be using `nocworx/phpcs` after we merge?

File: composer.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
   "repositories": [
     {
       "type": "vcs",
-      "url": "https://github.com/mdk999/coding-standard"
+      "url": "https://github.com/nocworx/symplify-coding-standard.git"
```
Comment (mclorch) [inline composer.json:17]: would this need a `composer.lock` change as well?

Comment (mclorch) [reply composer.json:17]: must have missed lock file changes completely when I wrote this

Comment (mclorch) [reply composer.json:17]: can be ignored

**Review by mclorch — APPROVED** [review]

**Comment by mpedrummer** [comment]
Superseded by #31

**Pull request closed without merging**